### ISSUE TITLE
New table creation fails

### ISSUE THREAD

**Issue by iiz00** [trey-wallis/obsidian-dataloom#604]
### Describe the bug

In v7.0.0, if the 'Create new dashboards in the attachments folder' option is turned on, new table creation fails. (If it is unchecked, it seems to work fine.)
When this happens, the Notice 'Could not create dashboad' appears on the screen, and when I look into the console,

I am seeing an error like below.
```
p.js:1 Uncaught (in promise) Error: Folder already exists.
    at t.<anonymous> (app.js:1:1024123)
    at app.js:1:236930
    at Object.next (app.js:1:237035)
    at a (app.js:1:235753)
```

Also, I don't know if this is relevant, I am seeing a number of errors:
```
bSocket connection to 'ws://localhost:8097/' failed:. 
connectToDevTools2 @ plugin:notion-like-tables:13314
eval @ plugin:notion-like-tables:13296
```

The settings on the Obsidian side are,
Default location for  new attachments: In subfolder under current folder
Subfolder name: attachments.

I didn't pay much attention to this problem in v6.x.x, but I think the same settings worked fine.

### Steps to reproduce

Turn 'Create new dashboards in the attachments folder' option on, and execute 'Dashboards: create dashboard' command.

### Expected behavior

New table is created.

### Are you using the mobile app?

No

### Obsidian debug info

SYSTEM INFO:
	Obsidian version: v1.3.7
	Installer version: v1.1.9
	Operating system: Windows 10 Pro 10.0.19045
	Login status: logged in
	Catalyst license: vip
	Insider build toggle: on
	Live preview: on
	Legacy editor: off
	Base theme: light
	Community theme: none
	Snippets enabled: 3
	Restricted mode: off
	Plugins installed: 45
	Plugins enabled: 18
		1: Recent Files v1.3.6
		2: Word Splitting for Japanese in Edit Mode v1.0.0
		3: MySnippets v1.2.3
		4: Obsidian42 - BRAT v0.7.0
		5: Snippet Commands v0.0.2
		6: Dashboards v7.0.0
		7: Outliner v4.7.1
		8: AutoMOC v1.1.1
		9: Customizable Menu v2.2.0
		10: Customizable Page Header and Title Bar v4.7.2
		11: Daily Note Outline v1.2.2
		12: Hotkeys for templates v1.4.3
		13: List Callouts v1.1.4
		14: Advanced URI v1.36.3
		15: Tasks v4.1.0
		16: Calendar v1.5.10
		17: Periodic Notes v0.0.17
		18: Various Complements v7.3.0

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

**Comment by trey-wallis** [comment]
@iiz00 Is this still happening in 7.0.1?

**Comment by iiz00** [comment]
Unfortunately, the problem seems to remain the same in v7.0.1.
- I can reproduce the problem even if I install only Dashboards in sandbox vault.
- The error message in console may have changed:
```
Uncaught (in promise) Error: File names cannot end with a dot or a space.
    at KE (app.js:1:1013977)
    at t.checkPath (app.js:1:1021397)
    at t.<anonymous> (app.js:1:1024058)
    at app.js:1:236930
    at Object.next (app.js:1:237035)
    at app.js:1:235951
    at new Promise (<anonymous>)
    at v (app.js:1:235696)
    at t.createFolder (app.js:1:1023948)
    at eval (plugin:notion-like-tables:56171:26)
```
- I have not seen 'connection failed' error messages so far in v7.0.1.
- I have changed the 'Default location for new attachments' option in Obsidian and it seems to work correctly only when I select 'In the folder specified below' and specify the folder to create directly.

**Comment by trey-wallis** [comment]
I was able to reproduce the "Folder already exists". This error is very odd:  "File names cannot end with a dot or a space"

**Comment by iiz00** [comment]
I checked, and when I tried it in the sandbox, I got a "dot or space" error, but in my main vault, I got the "Folder already exists" error. I think there is some configuration difference that leads to the difference in error messages. I don't see any difference in behavior outside of the console.

**Comment by trey-wallis** [comment]
Fixed in 8.1.0.

**Comment by remus-selea** [comment]
Might be related to this issue. The settings are:
- "Create new dashboards in the attachments folder" turned on
- "Default location for new attachments" is set to "Same folder as current file"

When creating an a loom with "Dataloom: Create loom and embed it into current file", it creates the loom file at the root instead of in the same folder as the current file.

**Comment by trey-wallis** [comment]
Could you make a new issue for this? This is a different problem. I haven't
implemented the current folder functionality yet.

**Comment by gmkado** [comment]
I'm getting the `Folder already exists` error trying to create a new loom with the `Create new looms in the attachments folder` selected on v8.1.0.

![image](https://github.com/trey-wallis/obsidian-dataloom/assets/6592135/b28d3897-aa99-4110-a6c9-709e98f2edb2)

**Comment by iiz00** [comment]
I have the same error, under the combination of 'Default location for new attachments: In subfolder under current folder' in Obsidian settings and 'Create new dashboards in the attachments folder' in DataLoom settings.
"In subfolder under current folder" is not implemented yet?

**Comment by trey-wallis** [comment]
@gmkado Fixed in 8.9.0
@iiz00 Added in 8.9.0
